code/datums/interactions/combat_mode.dm

```diff
@@ -19,12 +20,19 @@
 	if (!M.hud_used.has_interaction_ui)
 		return
 	var/atom/movable/screen/combattoggle/flashy/CT = new
+	held_hud = CT
 	CT.hud = H
 	CT.icon = H.ui_style
 	CT.combat_mode = src
 	UI = CT
 	return CT
 
+/datum/interaction_mode/combat_mode/Destroy(force, ...)
+	. = ..()
```

Comment:
this needs to be called last, so remove this and put `return ..()` at the end of the proc